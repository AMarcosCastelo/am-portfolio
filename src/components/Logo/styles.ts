import styled from 'styled-components'

export const Logo = styled.h1`
  display: inline-block;
  color: ${({ theme }) => theme.palette.text.main};
  font-family: ${({ theme }) => theme.font.family_3};

  position: fixed;
  left: 2rem;
  top: 2rem;
  z-index: 3;
`
