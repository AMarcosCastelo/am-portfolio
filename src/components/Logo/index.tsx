import * as S from './styles'

const Logo = () => <S.Logo>AM</S.Logo>

export default Logo
