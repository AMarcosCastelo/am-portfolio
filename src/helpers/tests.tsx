import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { LIGHT_THEME, DARK_THEME } from 'styles'

export const renderWithTheme = (
  children: React.ReactNode,
  theme?: 'light' | 'dark'
): RenderResult =>
  render(
    <ThemeProvider theme={theme === 'dark' ? DARK_THEME : LIGHT_THEME}>
      {children}
    </ThemeProvider>
  )
