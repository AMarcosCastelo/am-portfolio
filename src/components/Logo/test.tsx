import { screen } from '@testing-library/react'
import { renderWithTheme } from 'helpers'

import Logo from '.'

describe('<Logo />', () => {
  it('should render the Logo', () => {
    const { container } = renderWithTheme(<Logo />)

    expect(screen.getByRole('heading', { name: /AM/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
