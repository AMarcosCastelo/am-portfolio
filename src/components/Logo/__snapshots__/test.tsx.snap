// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the Logo 1`] = `
.c0 {
  display: inline-block;
  color: #000;
  font-family: 'Pacifico',cursive;
  position: fixed;
  left: 2rem;
  top: 2rem;
  z-index: 3;
}

<h1
  class="c0"
>
  AM
</h1>
`;
